cache_money/cache.py

```diff
@@ -0,0 +1,288 @@
+            )
+            return
+
+        keys = await self.keys(decode_to_str=False)
+        return await self.delete(keys)
+
+    async def keys(
+        self, function: Union[str, Callable] = "", decode_to_str: bool = True
+    ) -> Union[List[bytes], List[str]]:
+        """
+        Return keys from Redis that were cached by Cache Money. Note that if the parameter function is not provided and
+        Cache Money doesn't have a value for its prefix attribute, all keys in the Redis DB will be returned, even if
+        the keys were not set up by Cache Money.
+
+        Args:
+            function: If provided will only returns keys representing cache entries for the function.
+            decode_to_str: Whether the keys should be decoded before being returned to this function. If set to False
+                the function will return byte strings instead of strings.
+
+        Returns:
+            A list of string or byte strings representing the cache keys in Redis.
+        """
+        if not self.enabled:
+            return []
+
+        function = function.__name__ if not isinstance(function, str) else function
+
+        keys = await self.conn.keys(self.make_key(function) + "*")
+        if decode_to_str:
+            keys = [decode(key) for key in keys]
+        return keys
+
+    def _key_fn(a, k) -> str:
+        """Generate a hash from args and kwargs."""
+        return hashlib.md5(pickle.dumps((a, k))).hexdigest()
+
+    def cached(self, key_fn: Callable = _key_fn, timeout: int = None) -> Any:
+        """
+        Decorator that will transparently cache calls to the wrapped function. By default, the cache key will be made
+        up of the arguments passed in (like memoize), but you can override this by specifying a custom `key_fn`.
+
+        Usage:
+            from cache_money import cache_money
+            from cache_money.constants import CACHE_MINUTE
+
+            @cache.cached(timeout=CACHE_MINUTE)
+            async def addition(x, y):
+                return x + y
+
+            await addition(3, 4)  # Function is called.
+            await addition(3, 4)  # Not called, value is cached.
+            await add_numbers.bust(3, 4)  # Clear cache for (3, 4).
+            await add_numbers(3, 4)  # Function is called.
+
+        The decorated function also gains a new attribute named `bust` which will clear the cache for the given args.
+
+        Args:
+            key_fn: Function used to generate a key from the given args and kwargs.
+            timeout: Time to cache return values in seconds.
+
+        Returns:
+            Return the result of the decorated function call with the given args and kwargs.
+        """
+
+        def decorator(fn):
+            def make_key(args, kwargs):
+                """Apply the name of the function in front of the key, separated by a colon `:`."""
+                return f"{fn.__name__}:{key_fn(args, kwargs)}"
```

Comment:
A function's name is not really a signature. Many functions can have the same name. I would suggest putting the function's module (maybe part of it depending on how long the key can be) part of they key.

If we decide not to make the change, I suggest we document that.